src/classes/RD2_OpportunityEvaluation_BATCH.cls

```diff
@@ -58,15 +58,46 @@ public without sharing class RD2_OpportunityEvaluation_BATCH
         private set;
     }
 
-    /*******************************************************************************************************
+    /***
     * @description Count of RD's processed by the batch
     */
-    private Integer numberRDsProcessed = 0;
+    @TestVisible
+    private Integer recordsProcessed = 0;
 
-    /*******************************************************************************************************
+    /***
     * @description Count of RD's that hit errors while processing
     */
-    private Integer numberOfFailures = 0;
+    @TestVisible
+    private Integer recordsFailed = 0;
+
+    /**
+     * @description Contains current date value
+     */
+    private Date currentDate {
+        get {
+            if (currentDate == null) {
+                currentDate = Date.today();
+            }
+            return currentDate;
+        }
+        set;
+    }
+
+    /**
+     * @description Default constructor
+     */
+    public RD2_OpportunityEvaluation_BATCH() {
+        //Do not set the current date in constructor, Date.today() will be used by default.
+    }
+
+    /**
+     * @description Constructor setting the current date to the specified date.
+     * The constructor is mainly used for testing.
+     * @param currentDate Date considered as today
+     */
+    public RD2_OpportunityEvaluation_BATCH(Date currentDate) {
```

Comment:
@Julian88Tex , the batch has been changed to use a current date for testing via Apex anon window. However, note the batch should be scheduled and used in multi-day testing without providing date since that's how it will be scheduled and/or executed on demand on a client org.

This is example when batch is executed using specified date as "today":
```
Date currentDate = Date.newInstance(2020, 04, 01);
RD2_OpportunityEvaluation_BATCH batch = new RD2_OpportunityEvaluation_BATCH(currentDate);
Database.executeBatch(batch, 50);
```

Reply:
@heidmary ^